system/Database/Config.php

```diff
@@ -53,16 +55,21 @@ public static function connect($group = null, bool $getShared = true)
         if (is_array($group)) {
             $config = $group;
             $group  = 'custom-' . md5(json_encode($config));
-        }
+        } else {
+            /** @var DbConfig $dbConfig */
+            $dbConfig = config('Database');
 
-        $config ??= config('Database');
+            if ($group === null) {
```

Comment:
I'm in favor of getting rid `empty()` but we no longer handle `''` which I think is a possibility.

Reply:
Actually it looks like this is acceptable: https://3v4l.org/qSBFc

Reply:
The current `develop` code does not support empty string as `$group`.
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Config.php#L60-L62
`$group` will be the default group name.

And `Database::load()` does not accept empty string.
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Config.php#L78
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Database.php#L42-L46